ivy/functional/frontends/torch/tensor.py

```diff
@@ -694,13 +694,13 @@ def tril(self, diagonal=0):
     def index_select(self, dim, index):
         return torch_frontend.index_select(self._ivy_array, dim, index)
 
-    @with_unsupported_dtypes({"1.11.0 and below": ("bfloat16", "float16")}, "torch")
+    @with_unsupported_dtypes({"1.11.0 and below": ("float16", "complex")}, "torch")
     def clamp(self, min=None, max=None):
         if min is not None and max is not None and ivy.all(min > max):
```

Comment:
basically, there's two reasons for this:
1. In the documentation (under notes), it says that if the min is greater than the max, then it would set all the values to max. So i had to add this in the function.
2. if i didnt add this case, it would throw an error as some of the test cases have examples where min > max.

Here's the documentation Im referring to (its under notes): 
https://pytorch.org/docs/stable/generated/torch.clamp.html

Reply:
Thanks for the explanation! Here are my opinions:

1. The frontend method and instance method should have the same behavior. If you only add the logic to the instance method, the frontend method will behave differently when `min > max`,  this is not what we want! Also, the note you gave me is about the torch.clamp method, hence the logic for `min > max` should be handled in the frontend method, not the instance method.
If you still need to cover min > max cases, we should perform a `git blame` and ask the author of the clamp frontend method for changes instead, like what we are doing now
2. I tried to remove the `min > max` part of your code, and all the tests still passed. When I deep dive to reproduce your errors, I found that the errors were related to the hypothesis library, when it tried to sample integer arrays, then threw something like this:
```bash
hypothesis.errors.InvalidArgument: Cannot have max_value=-1 < min_value=1
```
This error will appear when you try to sample min array from range [-25, 0] and max array from range [1, 25]. I noticed that you have changed to sample min array from range [0, 25] and max array from range [26, 50], then the error disappears, like magic :)
3. Finally, ivy.all is not the right operator for this case, e.g. when you pass min = tensor([0.5, -0.5]) and max = tensor([-0.5, 0.5], ivy.all will return False and then call the clamp frontend method

Reply:
1. the first statement is true, i was going to get round to that but didn't have the time :'(, thanks for pointing that out. i was wondering why it wasn't implemented in the frontend method so i added it in the instance method then come back to it at a later date and bring this up! 
2. I see, i thought ivy.all was best when completing the clamp function which was why i implemented it! 
3. i dont remember changing the min/max array, i thought i kept it as min=[-25,0] and max=[1,25]. i would need to change that!

Reply:
i see that you're removing bfloat16! Please don't remove that as it has caused hypothesis errors regarding this! thank you.

Reply:
Can you give me the error details? I have tested it several times but I can't see any errors, maybe the hypothesis team has fixed that error!
As you can see in Ivy docs [here](https://lets-unify.ai/docs/ivy/overview/deep_dive/data_types.html#supported-and-unsupported-data-types), the decorator `@with_unsupported_dtypes` and `@with_supported_dtypes` are used to flag the data types which this particular function does and does not support respectively for the associated backend. If torch supports bfloat16 for clamp, we should not add bfloat16 as an unsupported dtype.

Reply:
E       RuntimeError: "clamp_scalar_cpu" not implemented for 'Half'

E       TypeError: expected TensorOptions(dtype=float, device=cpu, layout=Strided, requires_grad=false (default), pinned_memory=false (default), memory_format=(nullopt)) (got TensorOptions(dtype=c10::BFloat16, device=cpu, layout=Strided, requires_grad=false (default), pinned_memory=false (default), memory_format=(nullopt)))
E       Falsifying example: test_torch_instance_clamp(
E           dtype_and_x=(['bfloat16'], [array([[-1]], dtype=bfloat16)]),
E           as_variable=False,
E           native_array=False,
E           class_=ivy.functional.frontends.torch.tensor.Tensor,
E           method_name='clamp',
E       )
E       

this was the error i came across when i was working with clamp

Reply:
If you got `RuntimeError: "clamp_scalar_cpu" not implemented for 'Half'`, it seems that you're using float16/half on CPU which does not support many of the operators ([here](https://github.com/pytorch/pytorch/issues/67684)). `Half` represents float16, and `BFloat16` represents bfloat16.

Reply:
yh that makes sense since i was using cpu rather than gpu! I understood that it was regarding float16 but i was unsure if it was bfloat16 or just float16! Thanks :)